File: output/gemma4-e4b_IO_1/avatar/Python/Java/codeforces_11_A.java
import java.util.Scanner;

public class codeforces_11_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read N and D (assuming they are the first two integers)
        if (!scanner.hasNextInt()) {
            return;
        }
        int n = scanner.nextInt();
        int d = scanner.nextInt();

        // Read the array X (reading all remaining integers)
        java.util.ArrayList<Integer> xList = new java.util.ArrayList<>();
        while (scanner.hasNextInt()) {
            xList.add(scanner.nextInt());
        }
        
        int[] x = new int[xList.size()];
        for (int i = 0; i < xList.size(); i++) {
            x[i] = xList.get(i);
        }

        long m = 0;
        int q = 0;

        // Iterate through the array up to the second to last element
        for (int a = 0; a < x.length - 1; a++) {
            if (x[a] >= x[a + 1]) {
                // Calculate q: q = (x[a] - x[a+1]) / d + 1
                // Since all values are integers, standard integer division is used.
                q = (x[a] - x[a + 1]) / d + 1;
                
                // Update m
                m = m + q;
                
                // Update x[a+1]
                x[a + 1] = x[a + 1] + q * d;
            }
        }

        System.out.println(m);
        
        scanner.close();
    }
}

File: output/gemma4-e4b/avatar/Java/Python/codeforces_11_A.py
import sys

def solve():
    # Read all input data from stdin
    data = sys.stdin.read().split()
    
    if not data:
        return

    # N is the first element, D is the second
    n = int(data[0])
    d = int(data[1])
    
    # The remaining elements are the array nums
    nums = [int(x) for x in data[2:2+n]]
    
    result = 0
    
    # Iterate from the second element (index 1)
    for i in range(1, n):
        # Calculate a = nums[i-1] - nums[i]
        a = nums[i-1] - nums[i]
        
        if a >= 0:
            # Calculate steps: a / d + 1. Use integer division (//)
            steps = a // d + 1
            
            # Update result
            result += steps
            
            # Update nums[i]
            nums[i] += steps * d
            
    print(result)

if __name__ == "__main__":
    solve()